### PR TITLE
feat: add Planning Msg Adapter 

### PR DESCRIPTION
**Pull request by shulanbushangshu**
## Description
add Planning Msg Adpater anfd test

 
## Tests performed
add Planning Msg Adpater anfd test

 

## Effects on system behavior
Not applicable.

 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: system/autoware_auto_msgs_adapter/src/include/adapter_planning.hpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+#include <autoware_auto_planning_msgs/msg/trajectory.hpp>
+#include <autoware_planning_msgs/msg/trajectory.hpp>
+
+#include <string>
+
+namespace autoware_auto_msgs_adapter
+{
+using Trajectory_auto = autoware_auto_planning_msgs::msg::Trajectory;
+using Point_auto = autoware_auto_planning_msgs::msg::TrajectoryPoint;
```
Comment (isamu-takagi) [inline system/autoware_auto_msgs_adapter/src/include/adapter_planning.hpp:29]: According to the guidelines, use `UpperCamelCase` for type names.
```suggestion
using TrajectoryAuto = autoware_auto_planning_msgs::msg::Trajectory;
using PointAuto = autoware_auto_planning_msgs::msg::TrajectoryPoint;
```

File: system/autoware_auto_msgs_adapter/src/include/adapter_planning.hpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      node.get_logger(), "AdapterPlanning is initialized to convert: %s -> %s",
+      topic_name_source.c_str(), topic_name_target.c_str());
+  }
+
+protected:
+  Trajectory_auto convert(const Trajectory & msg_source) override
+  {
+    Trajectory_auto msg_auto;
+    msg_auto.header = msg_source.header;
+    Point_auto TrajectoryPoint_auto;
```
Comment (isamu-takagi) [inline system/autoware_auto_msgs_adapter/src/include/adapter_planning.hpp:50]: According to the guidelines, use `lower_snake_case` for variables.
```suggestion
    PointAuto trajectory_point_auto;
```

**Comment by isamu-takagi** [comment]
This problem causes the colcon test to fail, so we have to wait for the merge. https://github.com/autowarefoundation/autoware.universe/pull/5744

**Comment by isamu-takagi** [comment]
@shulanbushangshu Could you update the variable name? https://github.com/autowarefoundation/autoware.universe/pull/5814#discussion_r1419969187

**Review by isamu-takagi — APPROVED** [review]
LGTM

**Review by beginningfan — APPROVED** [review]

**Pull request merged**